File: app/src/main/java/com/example/simonastojanovic/memeapp/MainActivity.java
package com.example.simonastojanovic.memeapp;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private MemeAdapter memeAdapter;

    private int position = RecyclerView.NO_POSITION;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        recyclerView = (RecyclerView) findViewById(R.id.recycler_view);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);

        recyclerView.setLayoutManager(layoutManager);

        memeAdapter = new MemeAdapter(this);
        recyclerView.setAdapter(memeAdapter);

        UriDownloadAsyncTask uriDownloadAsyncTask = new UriDownloadAsyncTask();
        uriDownloadAsyncTask.execute();

    }

    private class UriDownloadAsyncTask extends AsyncTask<String, Void, ArrayList<Meme>> {
        @Override
        protected ArrayList<Meme> doInBackground(String... params) {
            String uri = QueryUtils.getMemeUri();
            return QueryUtils.fetchResultData(uri);
        }

        @Override
        protected void onPostExecute(ArrayList<Meme> result) {

            memeAdapter.addResults(result);



        }
    }






}

File: app/src/main/java/com/example/simonastojanovic/memeapp/memelist/view/MainActivity.kt
package com.example.simonastojanovic.memeapp.memelist.view

import android.arch.lifecycle.Observer
import android.os.Bundle
import android.view.View
import com.example.simonastojanovic.memeapp.R
import com.example.simonastojanovic.memeapp.app.view.BaseActivity
import com.example.simonastojanovic.memeapp.memelist.viewmodel.MemeViewModel
import com.example.simonastojanovic.memeapp.memelist.widget.MemeListWidget
import javax.inject.Inject

class MainActivity : BaseActivity() {

    @Inject
    lateinit var memeListWidget: MemeListWidget

    @Inject
    lateinit var memeViewModel: MemeViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        memeViewModel.getMemeApiList()
    }

    override fun provideLayout(): Int = R.layout.activity_main

    override fun provideViewModel() {
        memeViewModel.memeLiveData.observe(this, Observer { it?.let { memeListWidget.addResults(it) } })
    }

    override fun initWidgets(view: View) {
        memeListWidget.init(view)
    }

}
